src/Microsoft.AspNet.OData.Shared/Query/Expressions/PropertyContainer.cs

```diff
@@ -212,7 +219,7 @@ private class SingleExpandedProperty<T> : NamedProperty<T>
 
             public override object GetValue()
             {
-                return IsNull ? (object)null : Value;
+                return IsNull ? (object)null : (RawValue == null ? (object)null : Value);
```

Comment:
Easier to read as 
```suggestion
                return IsNull || RawValue == null ? (object)null : Value);
```

Reply:
Thank you, sounds good. I'll make the change when I'm finished with the failing tests.